swarm_copy/tools.py

```diff
@@ -0,0 +1,42 @@
+from typing import Any, ClassVar
+
+from openai.lib._tools import pydantic_function_tool
+from pydantic import BaseModel, Field
+
+
+class BaseTool(BaseModel):
+    name: str
+    description: str
+    input_schema: type[BaseModel]
+    metadata: dict[str, Any]
```

Comment:
Alternatively, we could do something like `metadata_schema: BaseModel` and basically have the same mechanism as the for the `input_schema`.

Why potentially useful? Full type safety and we explicitly define what the metadata should hold.

We should just make sure that once we define `ToolFooMetadata` to set `extra="ignore"` just so that we can do the `**context_variables`